broker/grpc/src/client.cc

```diff
@@ -99,6 +99,7 @@ client::~client() {
 void client::start_read(event_ptr& to_read, bool first_read) {
   StartRead(to_read.get());
   if (first_read) {
+    AddHold();
```

Comment:
In the documentation, they also speak about RemoveHold(), but this is not present in your code...

Reply:
added in case of communication failure